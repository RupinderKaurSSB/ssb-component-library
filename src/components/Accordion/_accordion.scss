.ssb-accordion {
  display: inline-grid;
  width: 100%;

  .accordion-header {
    @include reset-button;

    .button-grid {
      align-items: center;
      cursor: pointer;
      display: grid;
      grid-column-gap: $global-padding-size;
      grid-template-columns: auto 20px;
    }

    .expand-icon {
      color: $ssb-green-4;
    }

    .header-text {
      color: $ssb-dark-6;
    }

    .sub-header {
      color: $ssb-green-4;
    }

    &:hover {

      .header-text, .sub-header {
        color: $ssb-green-4;
      }
    }

  }

  &.with-sub-header {

    .button-grid {
      grid-template-columns: fit-content(100px) auto 20px;
    }

    .accordion-header {

      .header-text {
        grid-column-start: 2;
      }

      .sub-header {
        align-self: start;
        grid-column-start: 1;
      }
    }
  }

  .accordion-body {
    @include open-sans;
    color: $ssb-dark-6;
    font-size: 16px;
    padding: $global-padding-size 50px $global-padding-size $global-padding-size;

    &.closed {
      display: none;
      height: 0;
    }
  }
}
