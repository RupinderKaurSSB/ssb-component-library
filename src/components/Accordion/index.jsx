import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {Â ChevronDown, ChevronUp } from 'react-feather';

const Accordion = ({
	id, children, className, header, openByDefault, subHeader, tabIndex, withoutBorders,
}) => {
	const [isOpen, toggleOpen] = useState(openByDefault);
	return (
		<div
			id={id}
			className={`ssb-accordion${subHeader ? ' with-sub-header' : ''}${withoutBorders ? ' without-borders' : ''}${className ? ` ${className}` : ''}`}
		>
			<button
				className={`accordion-header ${isOpen ? 'open' : 'closed'}`}
				tabIndex={tabIndex}
				onClick={() => toggleOpen(!isOpen)}
			>
				<span className="button-grid">
					{subHeader && <span className="sub-header">{subHeader}</span>}
					<span className="header-text">{header}</span>
					{!isOpen && <ChevronDown className="expand-icon" size={20} />}
					{isOpen && <ChevronUp className="expand-icon" size={20} />}
				</span>
			</button>
			<div className={`accordion-body ${isOpen ? 'open' : 'closed'}`}>
				{children}
			</div>
		</div>
	);
};

Accordion.defaultProps = {
	openByDefault: false,
	tabIndex: 0,
};

Accordion.propTypes = {
	id: PropTypes.string,
	children: PropTypes.node,
	className: PropTypes.string,
	header: PropTypes.string,
	openByDefault: PropTypes.bool,
	subHeader: PropTypes.string,
	tabIndex: PropTypes.number,
	withoutBorders: PropTypes.bool,
};

export default Accordion;
