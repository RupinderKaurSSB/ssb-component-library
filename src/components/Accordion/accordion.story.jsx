import React from 'react';
import { storiesOf } from '@storybook/react';
import Accordion from './index';
import Divider from '../Divider';
import NestedAccordion from '../NestedAccordion';

const mockText = `This is paragraph text which explains the accordion,
 the rest of the text is just to fill it out and show the space it takes.`;

storiesOf('Accordion', module)
	.add('Default', () => (
		<div style={{ display: 'flex', marginTop: '10%' }}>
			<div className="col-lg-6">
				<Accordion primary header="This is a accordion title">
					This is paragraph text which explains the accordion, the rest of the text is just to fill it out and show the space it takes.
				</Accordion>
			</div>
		</div>
	))
	.add('With sub-header', () => (
		<div style={{ display: 'flex', marginTop: '10%' }}>
			<div className="col-lg-6">
				<Accordion primary subHeader="Tabell 1" header="This is a accordion title with a sub-header">
					This is paragraph text which explains the accordion, the rest of the text is just to fill it out and show the space it takes.
				</Accordion>
			</div>
		</div>
	))
	.add('With nested accordions', () => (
		<div style={{ display: 'flex', marginTop: '10%' }}>
			<div className="col-lg-6">
				<Accordion header="This is a secondary header">
					<div style={{ marginBottom: '10px' }}>{mockText}</div>
					<NestedAccordion header="A nested accordion header">
						This is paragraph text which explains the accordion, the rest of the text is just to fill it out and show the space it takes.
					</NestedAccordion>
				</Accordion>
			</div>
		</div>
	))
	.add('Multiple w/ sub-header', () => (
		<div style={{ display: 'flex', marginTop: '10%' }}>
			<div className="col-lg-6">
				<Divider className="mb-1" light />
				<Accordion primary subHeader="Tabell 1" header="This is a accordion title with a sub-header, her you can see the text in an extra row width a 6 column max lenght">
					This is paragraph text which explains the accordion, the rest of the text is just to fill it out and show the space it takes.
				</Accordion>
				<Divider className="mb-1 mt-1" light />
				<Accordion primary subHeader="Tabell 2" header="This is a accordion title with a sub-header">
					This is paragraph text which explains the accordion, the rest of the text is just to fill it out and show the space it takes.
				</Accordion>
				<Divider className="mb-1 mt-1" light />
				<Accordion primary subHeader="Tabell 3" header="This is a accordion">
					This is paragraph text which explains the accordion, the rest of the text is just to fill it out and show the space it takes.
				</Accordion>
				<Divider className="mt-1" light />
			</div>
		</div>
	));
