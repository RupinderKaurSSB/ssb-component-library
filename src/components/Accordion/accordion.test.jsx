import React from 'react';
import {shallow} from 'enzyme';
import Accordion from './index';

describe('Accordion component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Accordion header="This is a accordion title">Accordion</Accordion>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Toggles classNames correctly', () => {
		const wrapper = shallow(<Accordion header="This is a accordion title">This is paragraph text</Accordion>);
		expect(wrapper.find('.accordion-header').hasClass('closed')).toEqual(true);
		expect(wrapper.find('.accordion-body').hasClass('closed')).toEqual(true);
	});
	test('Change classname to open on click', () => {
		const wrapper = shallow(<Accordion header="This is a accordion title">This is paragraph text</Accordion>);
		wrapper.find('.accordion-header').simulate('click');
		expect(wrapper.find('.accordion-header').hasClass('open')).toEqual(true);
		expect(wrapper.find('.accordion-body').hasClass('open')).toEqual(true);
	});
	test('Accordion with sub-header', () => {
		const wrapper = shallow(<Accordion header="This is a accordion title" subHeader="Subheader">This is paragraph text</Accordion>);
		expect(wrapper.hasClass('with-sub-header')).toEqual(true);
		expect(wrapper.find('.sub-header').length).toBe(1);
	});
});
