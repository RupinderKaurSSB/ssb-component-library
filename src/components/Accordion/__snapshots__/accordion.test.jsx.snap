// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component Matches the snapshot 1`] = `
<div
  className="ssb-accordion undefined"
>
  <button
    className="accordion-header closed"
    onClick={[Function]}
    tabIndex={0}
  >
    <span
      className="button-grid"
    >
      <Title
        className="header-text no-margin"
        negative={false}
        size={5}
      >
        This is a accordion title
      </Title>
      <ChevronDown
        className="expand-icon"
        color="currentColor"
        size={20}
      />
    </span>
  </button>
  <div
    className="accordion-body closed"
  />
</div>
`;
